dev/com.ibm.ws.config/src/com/ibm/ws/config/xml/internal/SystemConfiguration.java

```diff
@@ -174,16 +188,17 @@ void start() throws ConfigUpdateException, ConfigValidationException, ConfigPars
             serverXMLConfig.loadInitialConfiguration(variableRegistry);
         }
 
-        final boolean reprocessConfig;
+        reprocessConfig();
+    }
+
+    void reprocessConfig() throws ConfigUpdateException {
         if (serverXMLConfig.isModified() || variableRegistry.variablesChanged()) {
```

Comment:
I'm unsure if the call to `variableRegistry.variablesChanged()` will notice that the symbol for `WsLocationConstants.LOC_VARIABLE_SOURCE_DIR` has changed on restore.

Reply:
I added a restore hook in ConfigVariableRegistry to update the registry if variable_source_dir changes. ConfigVariableRegistry.variablesChanged()` method compares the values in cache and registry. After my changes, the registry now has the updated values which will be compared to old cache entries.